setup.sh

```diff
@@ -1,3 +1,12 @@
+# xcode command line tools installation will hang on OS versions lower than this
+MIN_OS_VERSION="12.4.0"
+
+# use version sorting to check if the current version is less than $MIN_OS_VERSION
+if [[ $MIN_OS_VERSION != "$(printf "$MIN_OS_VERSION\n$(sw_vers -productVersion)" | sort -V | sed -n 1p)" ]]; then
+    echo "MacOS minimum required version is ${MIN_OS_VERSION}. Please update your OS before running this script."
```

Comment:
Thanks for the meaningful error message. Non-blocking suggestion: Maybe include the current OS version?

Reply:
Good suggestion. I've made this update which also cleaned up the code by storing the version check in a variable.